src/PrestaShopBundle/Controller/Admin/Sell/Address/AddressController.php

```diff
@@ -473,6 +476,99 @@ public function editOrderAddressAction(int $orderId, string $addressType, Reques
+    {
+        try {
+            // @todo: don't rely on Cart ObjectModel, use a Adapter DataProvider
+            $cart = new Cart($cartId);
+            $addressId = null;
+            switch ($addressType) {
+                case 'delivery':
+                    $addressType = CartAddressType::DELIVERY_ADDRESS_TYPE;
+                    $addressId = $cart->id_address_delivery;
+                    break;
+                case 'invoice':
+                    $addressType = CartAddressType::INVOICE_ADDRESS_TYPE;
+                    $addressId = $cart->id_address_invoice;
+                    break;
+            }
+
+            /** @var EditableCustomerAddress $editableAddress */
+            $editableAddress = $this->getQueryBus()->handle(new GetCustomerAddressForEditing((int) $addressId));
+
+            $addressFormBuilder = $this->get(
+                'prestashop.core.form.identifiable_object.builder.address_form_builder'
+            );
+            // Special cart handler
+            $addressFormHandler = $this->get(
+                'prestashop.core.form.identifiable_object.handler.cart_address_form_handler'
+            );
+
+            // Address type required for EditCartAddressCommand
+            $formData = [
+                'address_type' => $addressType,
+            ];
+            // Country needs to be preset before building form type because it is used to build state field choices
+            if ($request->request->has('customer_address') && isset($request->request->get('customer_address')['id_country'])) {
+                $formCountryId = (int) $request->request->get('customer_address')['id_country'];
+                $formData['id_country'] = $formCountryId;
+            }
+
+            // Address form is built based on address id to fill the data related to this address
+            $addressForm = $addressFormBuilder->getFormFor($addressId, $formData);
+            $addressForm->handleRequest($request);
+
+            // Form is handled based on Cart ID because that's the cart that needs update
+            $result = $addressFormHandler->handleFor($cartId, $addressForm);
+
+            if ($result->isSubmitted() && $result->isValid()) {
+                $this->addFlash('success', $this->trans('Update successful', 'Admin.Notifications.Success'));
+
+                if ($request->query->has('submitFormAjax')) {
+                    return $this->render(
+                        '@PrestaShop/Admin/Sell/Address/modal_create_success.html.twig',
+                        ['refreshCartAddresses' => 'false']
+                    );
+                }
+
+                return $this->redirectToRoute('admin_carts_view', ['cartId' => $cartId]);
+            }
+        } catch (Exception $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
+
+            return $this->redirectToRoute('admin_carts_view', ['cartId' => $cartId]);
```

Comment:
Why redirect to cart listing pages if an exception is thrown ?
I suggest removing this `redirectToRoute()`, let's show the merchant the nice error message added in flash bag

Reply:
I was about to do it but I realized it might a bit trickier than I thought 😅 Because the try/catch block includes too much code, so it can fail when we get the data, when the form is created, ...

The risk is that some variable may not be initialized when we try to render the page This is the same problem we had recently with PHPStan, I don't remember how we solved this? Multiple try/catch blocks no?

Reply:
Yes, multiple try/catch